docs/cmake.md

```diff
@@ -50,6 +50,57 @@ depending upon which variables are enabled/disabled.
 If you are using OpenDDS libraries that were built without the help of the
 `configure` script, the `config.cmake` file needs to be created manually.
 
+### Manually Creating config.cmake
+
+In order to work with config.cmake it is important to understand the purpose
+of its contained variables. Below is a list of the important variables with
+some information on their purpose.
+
+#### Optional OpenDDS Features
+When OpenDDS is built using the MPC build-system
+and optional features are specified, various C/C++; tao_idl; and opendds_idl
+compiler macros are set on dependent libraries or executables. These defines
+are set by MPC using the [dcps_optional_features.mpb](../MPC/config/dcps_optional_features.mpb)
+file.
+
+CMake-Based targets must also inherit the defined macros, so the functionality
+in `dcps_optional_features.mpb` is simulated using `options.cmake`
+and its associated `config.cmake`. As such, if features are enabled/disabled when
+compiling OpenDDS then it is paramount that the MPC features are also matched
+within `config.cmake` prior to compilation.
+
+This table outlines the feature-related variables and their _expected_ default
+values (generated when the configure script is run without arguments).
+If any feature is explicitly enabled/disabled in the OpenDDS build then the corresponding
+CMake variable below should be set to either _ON_ or _OFF_ depending on the desired state.
+
+| CMake Variable                   | Notes (from configure script output)    | Default |
+|---                               | ---                                     | ---     |
+|`OPENDDS_BUILT_IN_TOPICS`         | Built-in Topics                         | ON      |
+|`OPENDDS_CONTENT_FILTERED_TOPIC`  | ContentFilteredTopic (CS Profile)       | ON      |
+|`OPENDDS_CONTENT_SUBSCRIPTION`    | Content-Subscription Profile            | ON      |
```

Comment:
I am thinking of adding some brief notes on the Content-Subscription profile as if this gets disabled the flags are added to remove other related parts:

```
if (NOT OPENDDS_CONTENT_SUBSCRIPTION)
  _OPENDDS_APPEND_DEF(
    OPENDDS_NO_QUERY_CONDITION
    OPENDDS_NO_CONTENT_FILTERED_TOPIC
    OPENDDS_NO_MULTI_TOPIC
  )
endif()
```